py/torch_tensorrt/dynamo/conversion/impl/upsample.py

```diff
@@ -14,54 +14,35 @@ def upsample(
     source_ir: Optional[SourceIR],
     name: str,
     input: TRTTensor,
-    out_shape: Optional[Sequence[int]],
-    scale_factors: Optional[Sequence[float]],
-    resize_mode: str,
+    size: Optional[Sequence[int]],
+    scale_factor: Optional[Sequence[float]],
+    mode: str,
     align_corners: bool,
 ) -> TRTTensor:
-    resize_layer = ctx.net.add_resize(input)
-    # output size calculation
-    # Pytorch assumes that one of out_shape/scale_factor is None
-    # Pytorch assumes that dimensions match for out_shape/scale factor
-    if out_shape is not None:
-        resize_layer.shape = list(input.shape)[:2] + list(out_shape)
-    elif scale_factors is not None:
-        resize_layer.scales = [1.0, 1.0] + list(scale_factors)
-    else:
-        raise RuntimeError(
-            "At least one of out_shape and scale_factors should be specified."
-        )
+    layer = ctx.net.add_resize(input)
 
-    # interpolate mode
-    if resize_mode == "nearest" or None:
-        resize_layer.resize_mode = trt.InterpolationMode.NEAREST
-    elif resize_mode == "bilinear":
-        resize_layer.resize_mode = trt.InterpolationMode.LINEAR
-        if align_corners is None or not align_corners:
-            raise RuntimeError(
-                f"Interpolation works differently is align_corners is False for {resize_mode} mode in PyTorch and TensorRT."
-            )
+    if size is not None:
+        layer.shape = list(input.shape)[:2] + list(size)
     else:
-        raise RuntimeError(
-            f"Interpolation mode is {resize_mode} which is not supported by TensorRT."
-        )
+        layer.scales = [1.0, 1.0] + list(scale_factor)
 
-    if resize_mode == "nearest":
-        resize_layer.coordinate_transformation = (
-            trt.ResizeCoordinateTransformation.ASYMMETRIC
+    if mode == "nearest":
+        layer.resize_mode = trt.InterpolationMode.NEAREST
+        layer.coordinate_transformation = trt.ResizeCoordinateTransformation.ASYMMETRIC
+    elif mode in ("linear", "bilinear", "trilinear"):
+        layer.resize_mode = trt.InterpolationMode.LINEAR
+        layer.coordinate_transformation = (
+            trt.ResizeCoordinateTransformation.ALIGN_CORNERS
+            if align_corners
+            else trt.ResizeCoordinateTransformation.HALF_PIXEL
+        )
+    elif mode == "bicubic":
+        layer.resize_mode = trt.InterpolationMode.CUBIC
+        layer.coordinate_transformation = (
+            trt.ResizeCoordinateTransformation.ALIGN_CORNERS
+            if align_corners
+            else trt.ResizeCoordinateTransformation.HALF_PIXEL
         )
```

Comment:
I see that for the mode== ("linear", "bilinear", "trilinear") or "bicubic" the` layer.coordinate_transformation `trt.ResizeCoordinateTransformation.HALF_PIXEL is instead of `trt.ResizeCoordinateTransformation.ASYMMETRIC`. Could you point me to the test case below which correctly would illustrate the difference between the two transformations?

Reply:
linear, bilinear, trilinear and bicubic modes cannot use `ASYMMETRIC` when `align_corners=False`, otherwise the output would be wrong and differed a lot from PyTorch's result.